indico/modules/events/registration/controllers/management/invitations.py

```diff
@@ -105,6 +105,20 @@ def _checkParams(self, params):
         RHManageRegFormBase._checkParams(self, params)
         self.invitation = RegistrationInvitation.get_one(request.view_args['invitation_id'])
 
+
+class RHRegistrationFormDeleteInvitation(RHRegistrationFormInvitationBase):
+    """Delete a registration invitation"""
+
     def _process(self):
         db.session.delete(self.invitation)
         return jsonify_data(invitation_list=_render_invitation_list(self.regform))
+
+
+class RHRegistrationFormMarkInvitationAsDeclined(RHRegistrationFormInvitationBase):
+    """Mark a registration is declined by the invitee."""
+
+    def _process(self):
+        if self.invitation.state == InvitationState.pending:
+            self.invitation.state = InvitationState.declined
+            db.session.flush()
```

Comment:
I am not sure, however i think you don't have to `flush` here since a `flush` is issued after each successful request. @ThiefMaster please correct me if i am wrong